framework/src/modules/keys/keys_module.ts

```diff
@@ -38,4 +52,170 @@ export class KeysModule extends BaseModule {
 			numberOfSignatures: 0,
 		},
 	};
+
+	// eslint-disable-next-line class-methods-use-this
+	public async beforeTransactionApply({
+		stateStore,
+		transaction: {
+			asset,
+			assetType,
+			getSigningBytes,
+			id,
+			moduleType,
+			senderID,
+			senderPublicKey,
+			signatures,
+		},
+	}: TransactionApplyInput): Promise<void> {
+		const sender = await stateStore.account.get<AccountKeyAsset>(senderID);
+		const { networkIdentifier } = stateStore.chain;
+		const transactionBytes = getSigningBytes();
+
+		const transactionWithNetworkIdentifierBytes = Buffer.concat([
+			networkIdentifier,
+			transactionBytes,
+		]);
+
+		// This is for registration of multisignature that requires all signatures
+		if (moduleType === this.type && assetType === RegisterAssetType) {
+			const decodedAsset = codec.decode(KeysSchema, asset);
+			const { mandatoryKeys, optionalKeys } = decodedAsset as DecodedAsset;
+
+			// For multisig registration we need all signatures to be present
+			if (mandatoryKeys.length + optionalKeys.length + 1 !== signatures.length) {
+				throw new Error('There are missing signatures');
+			}
+
+			// Check if empty signatures are present
+			if (!signatures.every(signature => signature.length > 0)) {
+				throw new Error('A signature is required for each registered key.');
+			}
+
+			// Verify first signature is from senderPublicKey
+			validateSignature(senderPublicKey, signatures[0], transactionWithNetworkIdentifierBytes, id);
+
+			// Verify each mandatory key signed in order
+			validateKeysSignatures(
+				mandatoryKeys,
+				signatures.slice(1, mandatoryKeys.length + 1),
+				transactionWithNetworkIdentifierBytes,
+				id,
+			);
+
+			// Verify each optional key signed in order
+			validateKeysSignatures(
```

Comment:
🤔 comment says verify.. but its doing validate.. and code is same as above line

Reply:
Code was originally as this; I think it's using `verify` as a generic term that some kind of assertion is being done. It's not the same code though this one is doing the verification for `optionalKeys` instead of `mandatoryKeys`

Reply:
then maybe just change the comment to `Validate each optional key signed in order`? its only static check rite? if so then validate is correct. but minor comment, you can choose to ignore if you want.